tests/ttnn/unit_tests/operations/pool/test_rotate.py

```diff
@@ -0,0 +1,398 @@
+
+    comparison_passed = torch.allclose(torch_output_nhwc, ttnn_output_torch, atol=atol, rtol=rtol)
+    assert comparison_passed, f"Tensor size test failed for shape {input_shape}"
+
+
+@pytest.mark.parametrize("channels", [16, 32, 48, 64, 96, 128])
+def test_channel_alignment(device, channels):
+    """Test different channel sizes that meet alignment requirements."""
+    torch.manual_seed(0)
+
+    input_shape = (1, 32, 32, channels)
+    angle = 90.0
+    torch_input_nhwc = torch.randn(input_shape, dtype=torch.bfloat16)
+
+    # PyTorch reference using golden function
+    golden_function = ttnn.get_golden_function(ttnn.rotate)
+    torch_output_nhwc = golden_function(torch_input_nhwc, angle=angle)
+
+    # TTNN
+    ttnn_input = ttnn.from_torch(torch_input_nhwc, layout=ttnn.ROW_MAJOR_LAYOUT, device=device)
+    ttnn_output = ttnn.rotate(ttnn_input, angle=angle)
+    ttnn_output_torch = ttnn.to_torch(ttnn_output)
+
+    # 90-degree rotation should be exact
+    assert torch.equal(
+        torch_output_nhwc, ttnn_output_torch
+    ), f"90-degree rotation should be exact for {channels} channels"
+
+
+# ============================================================================
+# Memory Configuration Tests
+# ============================================================================
+
+
+@pytest.mark.parametrize("memory_config", [ttnn.DRAM_MEMORY_CONFIG, ttnn.L1_MEMORY_CONFIG])
+def test_memory_configs(device, memory_config):
+    """Test different memory configurations."""
+    torch.manual_seed(0)
+
+    input_shape = (1, 32, 32, 32)
+    angle = 45.0
+    torch_input_nhwc = torch.randn(input_shape, dtype=torch.bfloat16)
+
+    # PyTorch reference using golden function
+    golden_function = ttnn.get_golden_function(ttnn.rotate)
+    torch_output_nhwc = golden_function(torch_input_nhwc, angle=angle)
+
+    # TTNN with specific memory config
+    ttnn_input = ttnn.from_torch(
+        torch_input_nhwc, layout=ttnn.ROW_MAJOR_LAYOUT, device=device, memory_config=memory_config
+    )
+    ttnn_output = ttnn.rotate(ttnn_input, angle=angle)
+    ttnn_output_torch = ttnn.to_torch(ttnn_output)
+
+    # Use tolerances for diagonal rotation
+    atol, rtol = 5.0, 0.05
+    comparison_passed = torch.allclose(torch_output_nhwc, ttnn_output_torch, atol=atol, rtol=rtol)
+    assert comparison_passed, f"Memory config test failed for {memory_config}"
+
+
+def test_sharded_memory(device):
+    """Test rotation with sharded memory configuration."""
+    torch.manual_seed(0)
+
+    input_shape = (1, 32, 32, 32)
+    angle = 45.0
+    torch_input_nhwc = torch.randn(input_shape, dtype=torch.bfloat16)
+
+    # PyTorch reference using golden function
+    golden_function = ttnn.get_golden_function(ttnn.rotate)
+    torch_output_nhwc = golden_function(torch_input_nhwc, angle=angle)
+
+    # Create sharded memory config
+    shard_spec = ttnn.ShardSpec(
+        ttnn.CoreRangeSet({ttnn.CoreRange(ttnn.CoreCoord(0, 0), ttnn.CoreCoord(0, 0))}),
```

Comment:
this is a test cases that is testing sharding where whole input is on a single core.
We should check other shardings (BS and WS unlikely to work based on kernel code), and something more elaborate than a single core case

Reply:
also we shoudl try nd sharded shard spec